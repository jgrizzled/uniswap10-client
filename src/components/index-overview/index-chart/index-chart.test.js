import IndexChart from '.';

const component = <IndexChart indexByDate={mockData.indexByDate} />;

// smoke test
it('renders without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<Providers>{component}</Providers>, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', async () => {
  const wrapper = Enzyme.mount(component, { wrappingComponent: Providers });
  expect(wrapper).toMatchSnapshot();
});
