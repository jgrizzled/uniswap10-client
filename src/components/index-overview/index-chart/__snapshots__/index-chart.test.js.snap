// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<IndexChart
  indexByDate={
    Array [
      Object {
        "date": "2020-01-01",
        "index": 100,
      },
      Object {
        "date": "2020-01-02",
        "index": 101,
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-AxheI gAFYUI"
    >
      <styled.div>
        <div
          className="sc-AxjAm qfvsW"
        >
          <styled.h2>
            <h2
              className="sc-AxiKw dxaXnq"
            >
              Index
            </h2>
          </styled.h2>
          <ChartTop
            chartWindow={null}
            setChartWindow={[Function]}
          >
            <styled.div>
              <div
                className="sc-AxhCb garXwA"
              >
                <span>
                  Total Return:  
                  <Percent
                    number={0.010000000000000009}
                  >
                    <styled.span
                      color="#74954F"
                    >
                      <span
                        className="sc-AxirZ gacQIj"
                        color="#74954F"
                      >
                        +
                        1
                        %
                      </span>
                    </styled.span>
                  </Percent>
                </span>
                <span>
                  <WindowButton
                    checked={true}
                    onChange={[Function]}
                  >
                    <styled.label
                      checked={true}
                    >
                      <label
                        checked={true}
                        className="sc-AxgMl jnClsi"
                      >
                        <styled.input
                          checked={true}
                          onChange={[Function]}
                        >
                          <input
                            checked={true}
                            className="sc-AxhUy fYpZjv"
                            onChange={[Function]}
                            type="radio"
                          />
                        </styled.input>
                        <span>
                          All
                        </span>
                      </label>
                    </styled.label>
                  </WindowButton>
                  <WindowButton
                    checked={false}
                    onChange={[Function]}
                  >
                    <styled.label
                      checked={false}
                    >
                      <label
                        checked={false}
                        className="sc-AxgMl csyjLF"
                      >
                        <styled.input
                          checked={false}
                          onChange={[Function]}
                        >
                          <input
                            checked={false}
                            className="sc-AxhUy fYpZjv"
                            onChange={[Function]}
                            type="radio"
                          />
                        </styled.input>
                        <span>
                          1y
                        </span>
                      </label>
                    </styled.label>
                  </WindowButton>
                  <WindowButton
                    checked={false}
                    onChange={[Function]}
                  >
                    <styled.label
                      checked={false}
                    >
                      <label
                        checked={false}
                        className="sc-AxgMl csyjLF"
                      >
                        <styled.input
                          checked={false}
                          onChange={[Function]}
                        >
                          <input
                            checked={false}
                            className="sc-AxhUy fYpZjv"
                            onChange={[Function]}
                            type="radio"
                          />
                        </styled.input>
                        <span>
                          3m
                        </span>
                      </label>
                    </styled.label>
                  </WindowButton>
                  <WindowButton
                    checked={false}
                    onChange={[Function]}
                  >
                    <styled.label
                      checked={false}
                    >
                      <label
                        checked={false}
                        className="sc-AxgMl csyjLF"
                      >
                        <styled.input
                          checked={false}
                          onChange={[Function]}
                        >
                          <input
                            checked={false}
                            className="sc-AxhUy fYpZjv"
                            onChange={[Function]}
                            type="radio"
                          />
                        </styled.input>
                        <span>
                          1m
                        </span>
                      </label>
                    </styled.label>
                  </WindowButton>
                </span>
              </div>
            </styled.div>
          </ChartTop>
          <ResponsiveContainer
            aspect={1.3636363636363635}
            debounce={0}
            height="100%"
            width="100%"
          >
            <div
              className="recharts-responsive-container"
              id="undefined"
              style={
                Object {
                  "height": "100%",
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "width": "100%",
                }
              }
            >
              <LineChart
                barCategoryGap="10%"
                barGap={4}
                data={
                  Array [
                    Object {
                      "date": "2020-01-01",
                      "index": 100,
                    },
                    Object {
                      "date": "2020-01-02",
                      "index": 101,
                    },
                  ]
                }
                height={0}
                layout="horizontal"
                margin={
                  Object {
                    "bottom": 5,
                    "left": 5,
                    "right": 5,
                    "top": 5,
                  }
                }
                reverseStackOrder={false}
                stackOffset="none"
                width={0}
              />
              <ResizeDetector
                handleHeight={true}
                handleWidth={true}
                nodeType="div"
                onResize={[Function]}
                querySelector={null}
                refreshRate={1000}
                skipOnMount={false}
                targetDomEl={null}
              >
                <ChildWrapper>
                  <div />
                </ChildWrapper>
              </ResizeDetector>
            </div>
          </ResponsiveContainer>
        </div>
      </styled.div>
    </div>
  </styled.div>
</IndexChart>
`;
